src/TreeSitter/Importing.hs

```diff
@@ -1,135 +1,267 @@
--- WIP: Do not depend on this yet!
-{-# LANGUAGE DeriveFunctor, FlexibleContexts, GeneralizedNewtypeDeriving, ScopedTypeVariables, TypeApplications #-}
-module TreeSitter.Importing where
+{-# LANGUAGE DefaultSignatures, FlexibleContexts, FlexibleInstances,
+             PolyKinds, ScopedTypeVariables, TypeApplications, TypeOperators #-}
+module TreeSitter.Importing
```

Comment:
A note probably for a future PR: we should rename the module to something like `TreeSitter.Unmarshal` now.